RiotSwiftUI/Modules/Authentication/VerifyMsisdn/Test/Unit/AuthenticationVerifyMsisdnViewModelTests.swift

```diff
@@ -0,0 +1,58 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+
+@testable import RiotSwiftUI
+
+class AuthenticationVerifyMsisdnViewModelTests: XCTestCase {
+    private enum Constants {
+        static let counterInitialValue = 0
+    }
+    
+    var viewModel: AuthenticationVerifyMsisdnViewModelProtocol!
+    var context: AuthenticationVerifyMsisdnViewModelType.Context!
+    
+    override func setUpWithError() throws {
+        viewModel = AuthenticationVerifyMsisdnViewModel()
+        context = viewModel.context
+    }
+
+    func testSentSMSState() async {
+        // Given a view model where the user hasn't yet sent the verification email.
+        XCTAssertFalse(context.viewState.hasSentSMS, "The view model should start with hasSentSMS equal to false.")
+        
+        // When updating to indicate that an email has been send.
+        let task = Task { await viewModel.updateForSentSMS() }
+        _ = await task.result
+        
```

Comment:
A learning I've made that neatens up these tests (but I clearly haven't applied going back to older code) is to put the test on the main actor and then the task can go:
```suggestion
    @MainActor func testSentSMSState() async {
        // Given a view model where the user hasn't yet sent the verification email.
        XCTAssertFalse(context.viewState.hasSentSMS, "The view model should start with hasSentSMS equal to false.")
        
        // When updating to indicate that an email has been send.
        viewModel.updateForSentSMS()
        
```